tests/v2/validation/clustertemplates/provisioning/provisioning_clustertemplate_test.go

```diff
@@ -0,0 +1,286 @@
+	log.Info("Create an rke template and creating a downstream node driver with the rke template.")
+
+	clusterTemplate, err := clustertemplates.CreateRkeTemplate(ct.client, nil)
+	require.NoError(ct.T(), err)
+
+	templateRevisonConfig := mgmt.ClusterTemplateRevision{
+		ClusterConfig: &mgmt.ClusterSpecBase{RancherKubernetesEngineConfig: &mgmt.RancherKubernetesEngineConfig{
+			Network: &mgmt.NetworkConfig{Plugin: ct.provisioningConfig.CNIs[0]},
+			Version: ct.provisioningConfig.RKE1KubernetesVersions[0],
+		},
+		},
+	}
+
+	clusterTemplateRevision, err := clustertemplates.CreateRkeTemplateRevision(ct.client, templateRevisonConfig, clusterTemplate.ID)
+	require.NoError(ct.T(), err)
+
+	rke1Provider := provisioning.CreateRKE1Provider(ct.provisioningConfig.Providers[0])
+	nodeTemplate, err := rke1Provider.NodeTemplateFunc(ct.client)
+	require.NoError(ct.T(), err)
+
+	nodeAndRoles := []provisioninginput.NodePools{
+		provisioninginput.AllRolesNodePool,
+	}
+
+	clusterObj, err := provisioning.CreateProvisioningRKE1ClusterWithClusterTemplate(ct.client, rke1Provider, clusterTemplate.ID, clusterTemplateRevision.ID,
+		nodeAndRoles, nodeTemplate)
+	require.NoError(ct.T(), err)
+
+	log.Info("Verifying the rke1 cluster comes up active.")
+
+	clusterConfig := clusters.ConvertConfigToClusterConfig(ct.provisioningConfig)
+	clusterConfig.KubernetesVersion = ct.provisioningConfig.RKE1KubernetesVersions[0]
+	provisioning.VerifyRKE1Cluster(ct.T(), ct.client, clusterConfig, clusterObj)
+}
+
+func (ct *ClusterTemplateProvisioningTestSuite) TestClusterTemplateEnforcementAsAdmin() {
+	log.Info("Enforcing cluster template while provisioning rke1 clusters")
+
+	clusterEnforcement, err := ct.client.Steve.SteveType(settings.ManagementSetting).ByID(clustertemplates.ClusterEnforcementSetting)
+	require.NoError(ct.T(), err)
+	_, err = settings.UpdateGlobalSettings(ct.client.Steve, clusterEnforcement, clustertemplates.EnabledClusterEnforcementSetting)
+	require.NoError(ct.T(), err)
+
+	verifySetting, err := ct.client.Steve.SteveType(settings.ManagementSetting).ByID(clustertemplates.ClusterEnforcementSetting)
+	require.NoError(ct.T(), err)
+	require.Equal(ct.T(), verifySetting.JSONResp["value"], clustertemplates.EnabledClusterEnforcementSetting)
+
+	log.Info("Creating a cluster template and verifying the user is not able to create the cluster without the required permissions.")
+	clusterTemplate, err := clustertemplates.CreateRkeTemplate(ct.client, nil)
+	require.NoError(ct.T(), err)
+
+	templateRevisonConfig := mgmt.ClusterTemplateRevision{
+		ClusterConfig: &mgmt.ClusterSpecBase{RancherKubernetesEngineConfig: &mgmt.RancherKubernetesEngineConfig{
+			Network: &mgmt.NetworkConfig{Plugin: ct.provisioningConfig.CNIs[0]},
+			Version: ct.provisioningConfig.RKE1KubernetesVersions[0],
+		},
+		},
+	}
+
+	clusterTemplateRevision, err := clustertemplates.CreateRkeTemplateRevision(ct.client, templateRevisonConfig, clusterTemplate.ID)
+	require.NoError(ct.T(), err)
```

Comment:
Is this where you want an error to be required?

Reply:
no, that was in one of the below lines. I was showing an incorrect line on the call :)